### PR TITLE
[PLA-1826] Change beam events broadcast

### PR DESCRIPTION
**Pull request by leonardocustodio**
### **PR Type**
enhancement, dependencies, tests

___

### **Description**
- Removed the PHP CS Fixer configuration file.
- Replaced event broadcasting with `safeBroadcast` method in multiple files.
- Modified constructors and updated broadcast data structures in event classes.
- Adjusted conditional checks and spacing for consistency across various files.
- Reordered traits for consistency in multiple GraphQL mutation and type files.
- Updated dependencies to replace `php-cs-fixer` with `laravel/pint`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>.php-cs-fixer.php</strong><dd><code>Remove PHP CS Fixer configuration file</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.php-cs-fixer.php
- Removed the PHP CS Fixer configuration file.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-ef867390e21235c28e24595707bf636e1cc9cad29cf8ee4ac405c564a9facdf6">+0/-165</a>&nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement
</strong></td><td><details><summary>14 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>BatchProcess.php</strong><dd><code>Update event broadcasting and conditional spacing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Commands/BatchProcess.php
<li>Replaced event broadcasting with <code>safeBroadcast</code> method.<br> <li> Adjusted conditional spacing for consistency.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-343dcc3cf53276d2abb4f91c758dd786dd829a75de65f5d27ff508685b6598ca">+15/-8</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BeamBatchTransactionCreated.php</strong><dd><code>Update BeamBatchTransactionCreated event structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/BeamBatchTransactionCreated.php
<li>Modified constructor to accept mixed event data.<br> <li> Updated broadcast data structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-2c6774455f6f586ace29a867f5ba63d8b08df8d27638d9d697679618c2d206f4">+5/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BeamClaimPending.php</strong><dd><code>Update BeamClaimPending event structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/BeamClaimPending.php
<li>Modified constructor to accept mixed event data.<br> <li> Updated broadcast data structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-806005c062dedc7377821e9438b3317e516efdf5cae07fa0cd6166eee38dd652">+10/-9</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BeamCreated.php</strong><dd><code>Update BeamCreated event structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/BeamCreated.php
<li>Modified constructor to accept mixed event data.<br> <li> Updated broadcast data structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-ee1762ed029d9f3f2c55b0e703d3130a6739f167e6c22c8434aa2a6b40cf9ff2">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CreateBeamClaimsCompleted.php</strong><dd><code>Update CreateBeamClaimsCompleted event structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/CreateBeamClaimsCompleted.php
<li>Modified constructor to accept mixed event data.<br> <li> Updated broadcast data structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-ff9d6667397f522680105828c8dabe124d470d0ad0aa19e9b2034e83e5a598fd">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TokensAdded.php</strong><dd><code>Update TokensAdded event structure</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Events/TokensAdded.php
<li>Modified constructor to accept mixed event data.<br> <li> Updated broadcast data structure.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-26f22241fb4205d432bda01524245a2fab9a813c8dae889c8b35f477fc0a4e07">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AddTokensMutation.php</strong><dd><code>Adjust conditional checks in AddTokensMutation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/GraphQL/Mutations/AddTokensMutation.php
- Adjusted conditional checks for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-04de6236f1656ec423c21a96c9c3ed8a95a665b7fc17d616d1e4c31aa0c27f85">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ClaimBeamMutation.php</strong><dd><code>Reorder traits in ClaimBeamMutation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/GraphQL/Mutations/ClaimBeamMutation.php
- Reordered traits for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-f4080d41dd3f655ad87332d52f8518891a8259c17242a35f41dd2fd3c8756bbe">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CreateBeamMutation.php</strong><dd><code>Adjust conditional checks in CreateBeamMutation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/GraphQL/Mutations/CreateBeamMutation.php
- Adjusted conditional checks for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-7854b3fc25813803a4271cafe9cda209a7719d4c6493b48db43f6d02475e4b10">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UpdateBeamMutation.php</strong><dd><code>Adjust conditional checks in UpdateBeamMutation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/GraphQL/Mutations/UpdateBeamMutation.php
- Adjusted conditional checks for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-23e6650e27868b2ebc07e6fab45df00c6ad7914d28a55d96445bd1403b8c6a55">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BeamClaimType.php</strong><dd><code>Reorder traits in BeamClaimType</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/GraphQL/Types/BeamClaimType.php
- Reordered traits for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-7ecf5e220b5e25127da703c9bf15310b32abe2e271f0c21e932b9a516282abcb">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>BeamType.php</strong><dd><code>Adjust conditional checks in BeamType</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/GraphQL/Types/BeamType.php
- Adjusted conditional checks for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-53e71a165bbd916dd45d06a59976aa3aa5ca249d9732f5ae390b4de587f2179a">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IntegerRangeStringType.php</strong><dd><code>Reorder traits and adjust conditionals in IntegerRangeStringType</code></dd></summary>
<hr>

src/GraphQL/Types/Input/IntegerRangeStringType.php
<li>Reordered traits for consistency.<br> <li> Adjusted conditional checks for consistency.<br>

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-9194cdda850f5920257a85fd3f5b7213ab445b73f49613a5244e6428f7e0ab68">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ClaimBeam.php</strong><dd><code>Adjust exception handling in ClaimBeam job</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/Jobs/ClaimBeam.php
- Adjusted exception handling for consistency.

</details>
    

  </td>
  <td><a href="https://github.com/enjin/platform-beam/pull/73/files#diff-7d3aba416652164d8d70a96632010b3833033de1cd8b8bc965a72e135c8cc264">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by enjinabner — APPROVED** [review]
I think we need to do similar for multi-tenants to update the event test part

File: src/Commands/BatchProcess.php [hunk 1]
```diff
@@ -212,7 +213,13 @@ protected function processBatch(BeamType $type): int
                         'idempotency_key' => Str::uuid()->toString(),
                     ]);
                     BeamBatch::where('id', $batchId)->update(['transaction_id' => $transaction->id]);
-                    BeamBatchTransactionCreated::safeBroadcast($param['beamId'], $param['collectionId'], $transaction->id);
+                    BeamBatchTransactionCreated::safeBroadcast(
```
Comment (enjinabner) [inline src/Commands/BatchProcess.php:216]: Why is this modified? The `BeamBatchTransactionCreated::construct` params didn't change though?

**Review by enjinabner — CHANGES_REQUESTED** [review]
clarification

File: src/Events/BeamBatchTransactionCreated.php [hunk 2]
```diff
@@ -5,20 +5,21 @@
...
 use Enjin\Platform\Events\PlatformBroadcastEvent;
+use Illuminate\Database\Eloquent\Model;
 
 class BeamBatchTransactionCreated extends PlatformBroadcastEvent
 {
     /**
      * Creates a new event instance.
      */
-    public function __construct(string $beamId, string $collectionId, int $transactionId)
+    public function __construct(mixed $event, ?Model $transaction, ?array $extra)
```
Comment (enjinabner) [inline src/Events/BeamBatchTransactionCreated.php:15]: Mostly there's no data passed to `$transaction` and `$extra` better put default to `null`

File: src/Events/CreateBeamClaimsCompleted.php [hunk 3]
```diff
@@ -6,21 +6,21 @@
...
-use Illuminate\Support\Arr;
+use Illuminate\Database\Eloquent\Model;
 
 class CreateBeamClaimsCompleted extends PlatformBroadcastEvent
 {
     /**
      * Creates a new event instance.
      */
-    public function __construct(array $data)
+    public function __construct(mixed $event, ?Model $transaction, ?array $extra)
```
Comment (enjinabner) [inline src/Events/CreateBeamClaimsCompleted.php:16]: Mostly there's no data passed to $transaction and $extra better put default to null

File: src/Events/TokensAdded.php [hunk 4]
```diff
@@ -6,18 +6,21 @@
...
 use Illuminate\Broadcasting\Channel;
+use Illuminate\Database\Eloquent\Model;
 
 class TokensAdded extends PlatformBroadcastEvent
 {
     /**
      * Create a new event instance.
      */
-    public function __construct(array $data)
+    public function __construct(mixed $event, ?Model $transaction, ?array $extra)
```
Comment (enjinabner) [inline src/Events/TokensAdded.php:16]: Mostly there's no data passed to $transaction and $extra better put default to null

**Review by enjinabner — CHANGES_REQUESTED** [review]
improvement

File: src/Events/BeamBatchTransactionCreated.php (hunk 2 above)
Comment (enjinabner) [reply src/Events/BeamBatchTransactionCreated.php:15]: It'll lead to this error if params is not supplied
```
Too few arguments to function Enjin\Platform\Beam\Events\CreateBeamClaimsCompleted::__construct(), 1 passed in /Users/atudz/Documents/workspace/enjin-internal-platform/app/GraphQL/Queries/GetEventsQuery.php on line 229 and exactly 3 expected  
```

Comment (leonardocustodio) [reply src/Events/BeamBatchTransactionCreated.php:15]: PlatformBroadcastedEvent class handles that.... It defaults to null and pass the nulls to the constructor of the child class. (When calling safeBroadcast)

Is it not being handled?

File: src/Events/CreateBeamClaimsCompleted.php (hunk 3 above)
Comment (leonardocustodio) [reply src/Events/CreateBeamClaimsCompleted.php:16]: PlatformBroadcastedEvent class handles that.... It defaults to null and pass the nulls to the constructor of the child class. (When calling safeBroadcast)

Is it not being handled?

File: src/Events/TokensAdded.php (hunk 4 above)
Comment (leonardocustodio) [reply src/Events/TokensAdded.php:16]: PlatformBroadcastedEvent class handles that.... It defaults to null and pass the nulls to the constructor of the child class. (When calling safeBroadcast)

Is it not being handled?

File: src/Events/BeamBatchTransactionCreated.php (hunk 2 above)
Comment (enjinabner) [reply src/Events/BeamBatchTransactionCreated.php:15]: It's not working, I'm getting that exception above when only passing 1 parameter.

Comment (leonardocustodio) [reply src/Events/BeamBatchTransactionCreated.php:15]: That's the thing, you should not instantiate the event manualy, it should be called through safeBroadcast.

Comment (enjinabner) [reply src/Events/BeamBatchTransactionCreated.php:15]: I need to instantiate it manually to get the object instance and get its details. It's being used on internal platform
https://github.com/enjin/enjin-internal-platform/pull/185/files#diff-caa0ff5610635631a4670b843efd1a9b8ef5d6a5ff8538908fabbc0babb84d71R237

Comment (enjinabner) [reply src/Events/BeamBatchTransactionCreated.php:15]: No, it needs to get the actual class as I need to fetch the broadcastOn and broadcastWith data

Comment (leonardocustodio) [reply src/Events/BeamBatchTransactionCreated.php]: done

**Review by enjinabner — APPROVED** [review]

**Pull request merged**